nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -1093,6 +1133,19 @@ private Charset getCharsetFromMediaType(MediaType contentType) {
         return contentType != null ? contentType.charset(StandardCharsets.UTF_8) : StandardCharsets.UTF_8;
     }
 
+    /**
+     * Retrieve the directory in which OkHttp should cache responses. This method opts
+     * to use a temp directory to write the cache, which means that the cache will be written
+     * to a new location each time this processor is scheduled.
+     *
+     * Ref: https://github.com/square/okhttp/wiki/Recipes#response-caching
+     *
+     * @return the directory in which the ETag cache should be written
+     */
+    private static File getETagCacheDir() {
+        return Files.createTempDir();
```

Comment:
Should this directory be deleted on ``@OnUnscheduled`` annotation call? Or is this done automatically?

Reply:
As it's currently written, this should be done via OS-controlled temporary folder cleanup [1]. I elected to do this to 1) avoid manual cleanup and 2) avoid introducing something error prone. As this is only a cache, its durability isn't important.

[1] https://google.github.io/guava/releases/19.0/api/docs/com/google/common/io/Files.html#createTempDir()

Reply:
I think it's a good way to go. It should be platform agnostic and work as a sane default.

Reply:
Fair enough. I just thought that OS-controlled temporary folder was being cleaned at server reboot by default and a NiFi server is probably not supposed to be rebooted very often.

Reply:
@pvillard31 Since there is a configured size limit on the field, it should stick to a reasonable size since there is a sane default of 10MB. If a user is foolish enough to set that to 10GB, not our problem.

Reply:
@MikeThomsen - true but one temporary folder is created each time the processor is started. We could assume that the processor is going to be stopped/started a lot. But I agree, this is a edge case. I'm fine with the current implementation. If I really want to be meticulous I could suggest to add this information in the property description :)